### PR TITLE
Code quality improvements

### PR DESCRIPTION
**Pull request by HighCrit**
Fixed among other things potential memory leaks and concurrency issues

### CODE REVIEW

File: src/main/java/com/wynntils/core/framework/FrameworkManager.java [hunk 1]
```diff
@@ -52,6 +52,8 @@
         registeredOverlays.put(Priority.HIGHEST, new ArrayList<>());
     }
 
+    private FrameworkManager() {}
```
Comment (HeyZeer0) [inline src/main/java/com/wynntils/core/framework/FrameworkManager.java]: no reason to have a constructor for a class that's not instantiable.

File: src/main/java/com/wynntils/core/CoreManager.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import net.minecraftforge.common.MinecraftForge;
 
 public class CoreManager {
+    private CoreManager() {}
```
Comment (HeyZeer0) [inline src/main/java/com/wynntils/core/CoreManager.java]: no reason to have a constructor for a class that's not instantiable.

File: src/main/java/com/wynntils/Reference.java [hunk 3]
```diff
@@ -64,9 +64,9 @@ public static boolean inClassSelection() {
     public static boolean developmentEnvironment = false;
 
     public static class ServerIPS {
+        private ServerIPS() {}
```
Comment (HeyZeer0) [inline src/main/java/com/wynntils/Reference.java]: no reason to have a constructor for a class that's not instantiable.

**Review by HeyZeer0 — CHANGES_REQUESTED** [review]
LGTM once changes done.

Nice work 👍🏻

File: src/main/java/com/wynntils/core/CoreManager.java (hunk 2 above)
Comment (magicus) [reply src/main/java/com/wynntils/core/CoreManager.java]: Actually, you always have a default constructor unless you specify another explicitly.

This change is implementing the recommended pattern for non-instantiable classes like singletons.

**Review by magicus — APPROVED** [review]

File: src/main/java/com/wynntils/core/CoreManager.java (hunk 2 above)
Comment (HighCrit) [reply src/main/java/com/wynntils/core/CoreManager.java]: what he said

Comment (HeyZeer0) [reply src/main/java/com/wynntils/core/CoreManager.java]: still, it doesn't matter and just pollutes your view with a completely useless constructor.

**Review by HeyZeer0 — APPROVED** [review]

**Pull request merged**